plugins/mongo_db_plugin/mongo_db_plugin.cpp

```diff
@@ -742,7 +744,8 @@ void mongo_db_plugin_impl::_process_accepted_transaction( const chain::transacti
    using bsoncxx::builder::basic::make_array;
    namespace bbb = bsoncxx::builder::basic;
 
-   const signed_transaction& trx = t->packed_trx()->get_signed_transaction();
+   const packed_transaction_v0_ptr ptv0 = t->packed_trx()->to_packed_transaction_v0();
+   const signed_transaction& trx = ptv0->get_signed_transaction();
```

Comment:
This (rather than `const transaction&`) is only needed because of the call to `to_variant_with_abi` below. Just like I suggested in the `history_plugin`, can we manually construct the variant by converting the `transaction` first and then manually appending the `signatures` and `context_free_data`? That way we avoid the expensive call to `to_packed_transaction_v0`.

Reply:
I suggest we don't bother since this is likely not too much worse than the copy of the variant and `mongo_db_plugin` days are numbered.